src/org/daisy/dotify/formatter/impl/LazyFormatterContext.java

```diff
@@ -27,11 +27,13 @@ public LazyFormatterContext(
         this.config = config;
     }
 
-    public synchronized FormatterContext getFormatterContext() {
-        if (context == null) {
-            context = new FormatterContext(translatorFactory, tbf, config);
+    public FormatterContext getFormatterContext() {
+        synchronized (this) {
```

Comment:
Please use a `private final static Object` for synchronizing.

Reply:
To my knowledge, that will not add any extra thread safety. The only thing it will introduce is another object that uses memory.